Standards/scs-XXXX-vN-security-groups-decision-record.md

```diff
@@ -0,0 +1,184 @@
+---
+title: Security Groups Decision Record
+type: Decision Record
+status: Draft
+track: IaaS
+author: josephineSei
+date: DD-03-2024
+---
+
+## Introduction
+
+Security Groups in IaaS (OpenStack) are sets of ip table rules, that are applied to ports which connect a VM to a network.
+In contrast to other resources like flavors or volume types that are always publicly accessible, or images that can be both public and private, security groups are always bound to the project level.
+That creates some difficulties for a possible standard of Security Groups, which are discussed in this document.
+
+## Terminology
+
+Security Group
+  A set of iptables rules that is applied to ports connecting a VM and a network.
+
+Security Group Rule (abbr. Rule)
+  This references a single rule within a security group.
+
+RBAC
+  Role Based Access Control used for policies and alike.
+
+network rbac / neutron rbac
+  These access controls will let administrators and users share neutron related resources with other projects.
+
+admin
+  The most powerful role in OpenStack. Only given to some Operators of the Cloud Infrastructure.
+
+## Context
+
+While creating a VM and also later on, one or more security groups can be added to it.
+When there is no security group specified the default security group will always be added.
+Like every other security group, the default group is also project bound.
+That means, it can be edited as required by project members.
+By design of OpenStack and when not changed the default rules in the default security group block ALL incoming traffic and only allow outgoing traffic[^1].
```

Comment:
```suggestion
By design of OpenStack and when not changed, default rules in default security group block ALL incoming traffic and only allow outgoing traffic[^1].
```

Reply:
Is that true? I think the default traditionally allowed for incoming traffic originating from the SAME group.

Reply:
If we want to diverge from the traditional behavior, we need to communicate this will with our users.
An upstream discussion to align this between OpenStack providers even beyond SCS would also make sense.

Reply:
I haven't yet found time to look into the code, but what @garloff wrote during the call today seems to indicate that he's right.

Should we schedule another out of band meeting to discuss this, how do we want to handle this?

Reply:
I was relying on the Security Groups document from the Nova team: https://docs.openstack.org/nova/latest/user/security-groups.html (also mentioned in the DR). The second paragraph in there is: "All projects have a default security group which is applied to any port that has no other security group defined. Unless you change the default, this security group denies all incoming traffic and allows only outgoing traffic from your instance."

So maybe we do have an inconsistency in the Upstream Nova docs too.
But as this DR will only decide, whether there should be a set of pre-defined security groups or just a guide. This discussion belongs into the issue that will standardize the default security group rules: https://github.com/SovereignCloudStack/standards/issues/521 

Mittlerweile (seit bobcat) ist es tatsächlich default, dass auch ingress von einer bestimmten SG erlaubt ist:
https://github.com/openstack/neutron/blob/master/neutron/db/migration/alembic_migrations/versions/2023.2/expand/c33da356b165_security_group_default_rules.py

Ich werde diesen Abschnitt des DR umschreiben und im anderen Issue genauer darauf eingehen.
Es wäre auch gut zu wissen, wann (ab welchem release) sich der default von gar kein ingress zu ingress nur von derselben SG geändert hat.